src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -402,6 +416,91 @@ public function partialRefundAction(int $orderId, Request $request)
         ]);
     }
 
+    /**
+     * @param OrderForViewing $orderForViewing
+     */
+    private function handleOutOfStockProduct(OrderForViewing $orderForViewing)
+    {
+        $isStockManagementEnabled = $this->configuration->get('PS_STOCK_MANAGEMENT');
+        if (!$isStockManagementEnabled || $orderForViewing->isDelivered() || $orderForViewing->isShipped()) {
+            return;
+        }
+
+        foreach ($orderForViewing->getProducts()->getProducts() as $product) {
+            if ($product->getAvailableQuantity() <= 0) {
+                $this->addFlash(
+                    'warning',
+                    $this->trans('This product is out of stock: ', 'Admin.Orderscustomers.Notification') . ' ' . $product->getName()
```

Comment:
The placeholder must be in the string, something like `This product is out of stock: [1]` To be confirmed by @LouiseBonnard

Reply:
I just checked and it seems it was used exactly like this in the legacy.